ferry.go

```diff
@@ -890,6 +908,21 @@ func (f *Ferry) ReportProgress() {
 	}
 }
 
+func (f *Ferry) ReportState() {
+	callback := f.Config.StateCallback
+	state, err := f.SerializeStateToJSON()
```

Comment:
I think this is fine, but maybe it's worth mentioning that this method will lock the StateTracker which will in turn block the BinlogWriter and BatchWriter from proceeding. I'm pretty sure this method is quite fast, but just as a FYI.

Reply:
Good catch. I can add that note. Would you rather see it here or in the config where the interval is configured?

Reply:
Maybe here is fine. I don't if/how that lock will actually impact the run.